File: 08-DesignPatterns-Java/Solution/Strategy/src/Ducks/DecoyDuck.java
package Ducks;

import Fly.NoneFlyStrategy;
import Quack.RoboticQuackStrategy;

public class DecoyDuck extends Duck {
    public DecoyDuck() {
        super(new RoboticQuackStrategy(), new NoneFlyStrategy());
    }

    @Override
    public void display() {
        System.out.println("Looks like mallard duck, but made from metal.");
    }
}

File: 09-DesignPatterns-CSharp/09-2-2-Adapter/Ducks/DecoyDuck.cs
namespace Adapter.Ducks
{
    using System;
    using Adapter.Fly;
    using Adapter.Quack;

    public sealed class DecoyDuck : Duck
    {
        public DecoyDuck()
            : base(new RoboticQuackStrategy(), new NoneFlyStrategy())
        {
        }

        public override void Display() => Console.WriteLine("Looks like mallard duck, but made from metal.");
    }
}